src/jdk.management.agent/share/conf/management.properties

```diff
@@ -302,5 +302,23 @@
 #   If the pattern ends with "*", it matches any class with the pattern as a prefix.
 #   If the pattern is equal to the class name, it matches.
 #   Otherwise, the status is UNDECIDED.
-com.sun.management.jmxremote.serial.filter.pattern=java.lang.*;java.math.BigInteger;java.math.BigDecimal;java.util.*;javax.management.openmbean.*;javax.management.ObjectName;java.rmi.MarshalledObject;javax.security.auth.Subject;!*
+com.sun.management.jmxremote.serial.filter.pattern=\
+java.lang.*;\
+java.lang.reflect.Proxy;\
+java.math.BigInteger;
+java.math.BigDecimal;
+java.util.*;\
+javax.management.*;\
+javax.management.modelmbean.*;\
+javax.management.monitor.*;\
+javax.management.openmbean.*;\
+javax.management.relation.*;\
+javax.management.remote.*;\
+javax.management.remote.rmi.*;\
+javax.management.timer.*;\
+javax.rmi.ssl.*;\
+java.rmi.MarshalledObject;\
+java.rmi.dgc.*;\
+java.rmi.server.*;\
+javax.security.auth.Subject;!*
```

Comment:
What does the ! at the end indicate?

Reply:
It's "not everything else".

https://docs.oracle.com/en/java/javase/19/core/serialization-filtering1.html

"If a class name doesn’t match any filter, then it is allowed. If you want to allow only certain class names, then your filter must reject everything that doesn’t match. To reject all class names other than those specified, include !* as the last pattern in a class filter."

Reply:
Ok. It would be good to clarify that in the comment above this filter. Also, maybe put it on a new line. Otherwise at first glance it appears to have a relationship to the class immediately before it.

Does this mean that this filter list would serve no purpose if the !* was omitted? I'm just curious as to why the !* is needed rather than it just being default behavior that a class has to match a filter in the list.

Reply:
If a class is not matched, it is "undecided", mentioned at the end of the long comment.  That is not a rejection, which that existing long comment does not state.  For an actual rejection, we need the ! to match, so patterns generally end in !*

It's the same or very similar comment as in conf/security/java.security

I added a note about the !* at the end to clarify, as it is new to use the filter in this area, and yes put it on a new line.